src/iuse.cpp

```diff
@@ -8493,13 +8493,13 @@ int iuse::saw_barrel( player *p, item *, bool, const tripoint& )
         return 0;
     }
 
-    auto filter = [&]( const item& e ) {
+    auto filter = []( const item& e ) {
         if( !e.is_gun() || e.type->gun->barrel_length <= 0 ) {
             return false;
         }
         // cannot saw down barrel of gun that already has a barrel mod
-        return std::none_of( e.contents.begin(), e.contents.end(), [&]( const item& mod ) {
-            return mod.type->gunmod->location == "barrel";
+        return std::none_of( e.contents.begin(), e.contents.end(), []( const item& mod ) {
```

Comment:
This correctly identifies the problem however storing the result of item::gunmods() and then using this for `std::none_of` is preferred to directly accessing the `contents` vector.